docs/typescript/reference/dbos-class.md

```diff
@@ -0,0 +1,971 @@
+
+  // Initialize an instance, see "Instance Methods" below
+  async initialize() {
+    return Promise.resolve();
+  }
+}
+
+// Create and register an object instance for executing workflows
+const workflowInstance = DBOS.configureInstance(Workflows, 'InstanceA');
+```
+
+#### Calling Workflow Functions
+Functions that are registered as workflows with `@DBOS.workflow` can be called directly like regular async functions, but will receive DBOS execution guarantees.
+
+```typescript
+// Run functions directly
+const result1 = await Workflows.processWorkflow("value1");
+const result2 = await workflowsInstance.processInstanceWorkflow("value2");
+```
+
+##### Starting Background Workflows
+Workflow functions can also be started asynchronously using `DBOS.startWorkflow`.  When run with `DBOS.startWorkflow`, the function will run in the background, and a `WorkflowHandle` is returned from `DBOS`.  This [handle](../reference/workflow-handles.md) can be used to [interact with the workflow](#interacting-with-workflows) or await its result with `getResult()`.
+
+```typescript
+DBOS.startWorkflow<T extends ConfiguredInstance>(targetInstance: T, params?: StartWorkflowParams): InvokeFunctionsAsyncInst<T>;
+DBOS.startWorkflow<T extends object>(targetClass: T, params?: StartWorkflowParams): InvokeFunctionsAsync<T>;
+
+interface StartWorkflowParams {
+  workflowID?: string;
+  queueName?: string;
+}
+```
+
+The optional `StartWorkflowParams` parameter allows the following to be controlled:
+* workflowID: The [workflow ID](#assigning-workflow-ids) for the started workflow
+* queueName: The name of the [workflow queue](#using-workflow-queues) for the workflow
+
+Examples:
+```typescript
+// Run static functions asynchronously and get the result later
+const handle3 = await DBOS.startWorkflow(Workflows).processWorkflow("value3");
+// ...
+const result3 = await handle3.getResult();
+
+// Run instance functions asynchronously and get the result later
+const handle4 = await DBOS.startWorkflow(workflowInstance).processInstanceWorkflow("value4");
+// ...
+const result4 = await handle4.getResult();
+
+// Run static functions asynchronously on a queue, with an assigned ID, and get the result later
+const handle5 = await DBOS.startWorkflow(Workflows, {workflowID: 'unique wf id', queueName: wfq.name})
+  .processWorkflow("value5");
+// ...
+const result5 = await handle3.getResult();
+
+```
+
+##### Using Workflow Queues
+By default, the `startWorkflow` method [described above](#starting-background-workflows) always starts the target workflow immediately.  If it is desired to control the concurrency or rate of workflow invocations, a [`queue`](../reference/workflow-queues.md) can be specified.
+
+```typescript
+DBOS.withWorkflowQueue<R>(wfq: string, callback: ()=>Promise<R>) : Promise<R>
+```
+
+The `DBOS.withWorkflowQueue` function runs the provided `callback` function and returns its result.  Any workflows started within `callback` will be run on the queue named by `wfq`.
```

Comment:
Just to be clear, this is being replaced with the new interface from https://github.com/dbos-inc/dbos-transact-ts/pull/673, right?

Reply:
Augmented.

Reply:
I mean, the whole neat thing about v2 is that it just looks like function calls.  That can now use queues to throttle the app pace, which is pretty cool if you think about it.

Reply:
I honestly don't understand the callback-based queues interface.  I can see how to use it, but I don't see why it needs a callback. The new interface based on `DBOS.startWorkflow` makes a lot more sense. Are there advantages to the callback-based interface?

Reply:
Placed next to each other for a `startWorkflow`, no, it doesn't make much sense.
Without the startWorkflow, it's a wash, because you have the callback or you have the `await handle.getResult`
Placed apart, say as a middleware, or administrative policy, then could you see it?  Because that's where I see the value.  Run the workflows in this class on this queue, or on these HTTP routes in that queue, etc.

Reply:
I can see how you could use it, but I don't think it's enough to justify having two ways of doing it. Would rather keep things simple and support/document only the simpler/easier-to-understand way of using queues.